File: java/DE-EIG/src/deeig/Bincrossover.java
package deeig;

public class Bincrossover {

	public static void run(double[] u, double[] x, double[] v, double CR, int D) {
		// Binomial Crossover (10)
		int jrand = (int) Math.floor(Math.random() * D);
		for (int j = 0; j < D; ++j) {
			if (Math.random() < CR || j == jrand) {
				u[j] = v[j];
			} else {
				u[j] = x[j];
			}
		}
	}
}

File: matlab/bincrossover.m
function u = bincrossover(x, v, u, CR, D)
% BINCROSSOVER Binomial Crossover (10)
jrand = floor(1 + D * rand);
for i = 1 : D
	if rand < CR || i == jrand
		u(i) = v(i);
	else
		u(i) = x(i);
	end
end
end
